llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -4005,6 +4017,38 @@ bool SPIRVInstructionSelector::loadVec3BuiltinInputID(
   return Result && MIB.constrainAllUses(TII, TRI, RBI);
 }
 
+// Generate the instructions to load 32-bit integer builtin input IDs/Indices.
+// Like LocalInvocationIndex
+bool SPIRVInstructionSelector::loadBuiltinInputID(
+    SPIRV::BuiltIn::BuiltIn BuiltInValue, Register ResVReg,
+    const SPIRVType *ResType, MachineInstr &I) const {
```

Comment:
Why do you pass in `ResType` if you will not use it?

Reply:
Could we just use this type instead of building using U32Type? Then the same function becomes more general, with no extra code.

Reply:
Good idea, done.